src/k8s/pkg/k8sd/types/cluster_config.go

```diff
@@ -112,4 +173,107 @@ func ClusterConfigFromBootstrapConfig(b *apiv1.BootstrapConfig) ClusterConfig {
 			Port: b.K8sDqlitePort,
 		},
 	}
+
+	for _, component := range b.Components {
+		switch component {
+		case "network":
+			config.Network = Network{
+				Enabled: &[]bool{true}[0],
+			}
+		case "dns":
+			config.DNS = DNS{
+				Enabled: &[]bool{true}[0],
+			}
+		case "local-storage":
+			config.LocalStorage = LocalStorage{
+				Enabled: &[]bool{true}[0],
+			}
+		case "ingress":
+			config.Ingress = Ingress{
+				Enabled: &[]bool{true}[0],
+			}
+		case "gateway":
+			config.Gateway = Gateway{
+				Enabled: &[]bool{true}[0],
+			}
+		case "metrics-server":
+			config.MetricsServer = MetricsServer{
+				Enabled: &[]bool{true}[0],
+			}
+		case "load-balancer":
+			config.LoadBalancer = LoadBalancer{
+				Enabled: &[]bool{true}[0],
+			}
+		}
+	}
+
+	return config
+}
+
+func ClusterConfigFromUserFacing(ufConfig *apiv1.UserFacingClusterConfig) ClusterConfig {
```

Comment:
I would also include some validation here. I see the TODO for later, I think that's also OK as long as we don't forget about it